tests/components/apcupsd/test_config_flow.py

```diff
@@ -0,0 +1,216 @@
+        # the creation due same host / port.
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_USER},
+            data=CONF_DATA,
+        )
+        assert result["type"] == FlowResultType.ABORT
+        assert result["reason"] == "already_configured"
+
+        # Then, we create the integration once again using a different port. However,
+        # the apcaccess patch is kept to report the same serial number, we should
+        # reject the creation as well.
+        another_host = {
+            CONF_HOST: CONF_DATA[CONF_HOST],
+            CONF_PORT: CONF_DATA[CONF_PORT] + 1,
+        }
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_USER},
+            data=another_host,
+        )
+        assert result["type"] == FlowResultType.ABORT
+        assert result["reason"] == "already_configured"
+
+        # Now we change the serial number and add it again. This should be successful.
+        another_device_status = copy(MOCK_STATUS)
+        another_device_status["SERIALNO"] = MOCK_STATUS["SERIALNO"] + "ZZZ"
+        mock_parse.return_value = another_device_status
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_USER},
+            data=another_host,
+        )
+        assert result["type"] == FlowResultType.CREATE_ENTRY
+        assert result["data"] == another_host
+
+
+async def test_flow_works(hass: HomeAssistant) -> None:
+    """Test successful creation of config entries via user configuration."""
+    with patch("apcaccess.status.parse", return_value=MOCK_STATUS), patch(
+        "apcaccess.status.get", return_value=b""
+    ), _patch_setup() as mock_setup:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={CONF_SOURCE: SOURCE_USER},
+        )
+        assert result["type"] == FlowResultType.FORM
+        assert result["step_id"] == SOURCE_USER
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=CONF_DATA
+        )
+        await hass.async_block_till_done()
+        assert result["type"] == FlowResultType.CREATE_ENTRY
+        assert result["title"] == MOCK_STATUS["MODEL"]
+        assert result["data"] == CONF_DATA
+
+        mock_setup.assert_called_once()
+
+
+async def test_flow_minimal_status(hass: HomeAssistant) -> None:
+    """Test successful creation of config entries via user configuration when minimal status is reported."""
+    with patch("apcaccess.status.parse") as mock_parse, patch(
+        "apcaccess.status.get", return_value=b""
+    ), _patch_setup() as mock_setup:
+        # We test different combinations of minimal statuses, where the title of the
+        # integration will vary.
+        extras = (
```

Comment:
I'd move these items to a parameter and use `pytest.mark.parametrize`.

https://docs.pytest.org/en/stable/how-to/parametrize.html

Reply:
Agreed! This made this test much cleaner. 👍